src/serialized-utilities.cpp

```diff
@@ -0,0 +1,160 @@
+    {
+        _schema_version = get_value(_root, "schema version").get<std::string>();
+        _device_info = read_device_info();
+        _parameters = &_root["parameters"];
+    }
+    else
+    {
+        _parameters = &_root;
+    }
+}
+
+device_info json_preset_reader::read_device_info() const
+{
+    device_info info;
+    auto device = get_value( _root, "device" );
+    if (!device.is_null())
+    {
+        auto device_name = get_value(device, "name");
+        info.name = device_name.is_null() ? "" : device_name.get< std::string >();
+
+        auto product_line = get_value(device, "product line");
+        info.product_line = product_line.is_null() ? "" : product_line.get< std::string >();
+
+        auto fw_version = get_value(device, "fw version");
+        info.fw_version = fw_version.is_null() ? "" : fw_version.get< std::string >();
+    }
+    return info;
+}
+
+void json_preset_reader::check_device_info( const device_interface & device ) const
+{
+    // Looks for device information
+    if( ! _device_info.product_line.empty()
+        && ! compare_device_info_field( device,
+                                        _device_info.product_line,
+                                        RS2_CAMERA_INFO_PRODUCT_LINE ) )
+    {
+        throw librealsense::invalid_value_exception(
+            "preset product line does not match the connected device" );
+    }
+
+    if( ! _device_info.name.empty()
+        && ! compare_device_info_field( device, _device_info.name, RS2_CAMERA_INFO_NAME ) )
+    {
+        throw librealsense::invalid_value_exception(
+            "preset device name does not match the connected device" );
+    }
+
+    if( ! _device_info.fw_version.empty()
+        && device.supports_info( RS2_CAMERA_INFO_FIRMWARE_VERSION ) )
+    {
+        // If we got a FW version but not a product name or FW version is smaller than the device FW
+        // version
+        // --> Throw
+        if( _device_info.product_line.empty()
+            || firmware_version( _device_info.fw_version )
+                   < firmware_version( device.get_info( RS2_CAMERA_INFO_FIRMWARE_VERSION ) ) )
+            throw librealsense::invalid_value_exception(
+                "preset device FW version is lower than the connected device FW version" );
+    }
+}
+
+json json_preset_reader::get_value(json j, const std::string& field_key) const
+{
+    auto val_it = j.find(field_key);
+    if (val_it != j.end())
+    {
+        return val_it.value();
+    }
+    
+    return json(); // if no value found return an empty object
+}
+
+bool json_preset_reader::compare_device_info_field(const device_interface& device, const std::string& file_value, rs2_camera_info camera_info) const
+{
+    if (device.supports_info(camera_info))
+    {
+        return file_value == device.get_info(camera_info);
+    }
+    return false;
+}
+
+bool json_preset_reader::validate_schema() const
+{
+    auto schema_version_found = _root.find("schema version") != _root.end();
```

Comment:
What if the schema is `2.0`? Or 'X'?

Reply:
Currently I didn't enforce the schema version, cause it looks like if we will increase it to 2.0 we will need to support 1.0 :)
It's just there for documentation.

Should I enforce it?